zerver/lib/test_helpers.py

```diff
@@ -203,14 +203,65 @@ def __init__(self, post_data, user_profile):
         self._log_data = {} # type: Dict[str, Any]
         self.META = {'PATH_INFO': 'test'}
 
+INSTRUMENTING = True # TODO: make this False by default and require a cmd-line arg to turn it on
+INSTRUMENTED_CALLS = []
+
+def instrument_url(f):
+    # TODO: If we can know that INSTRUMENTING is False
+    #       early enough in the process of loading up
+    #       the tests and applying these decorators,
+    #       then we should have the decorator simply
+    #       return `f` for the non-instrumented case,
+    #       to keep tracebacks simple and avoid a
+    #       minor performance penalty.  Either that,
+    #       or do more of a monkey-patching approach
+    #       where the decorator just notes the function
+    #       is instrumentable and some other piece
+    #       of code monkey-patches it later.
+    #
+    def wrapper(self, url, info={}, **kwargs):
+        if INSTRUMENTING:
+            start = time.time()
+            result = f(self, url, info, **kwargs)
+            delay = time.time() - start
+            test_name = self.id()
+            INSTRUMENTED_CALLS.append(dict(
+                url=url,
+                status_code=result.status_code,
+                method=f.__name__, # TODO: use more concise names
+                delay=delay,
+                info=info,
+                test_name=test_name,
+                kwargs=kwargs))
+            return result
+        else:
+            return f(self, url, info, **kwargs)
+    return wrapper
+
+def write_instrumentation_report():
+    if INSTRUMENTING:
+        fn = 'var/url_coverage.txt' # TODO make sure path is robust here
+        with open(fn, 'w') as f:
+            for call in INSTRUMENTED_CALLS:
+                try:
+                    line = ujson.dumps(call)
+                    f.write(line + '\n')
+                except OverflowError:
```

Comment:
I figured out what was causing the OverflowError, and it's fixed in #1448 (pending), so I am going to remove this commit from the next push, and it will fail hard in the future.  (There's no danger of it failing hard, since it's just tests.)